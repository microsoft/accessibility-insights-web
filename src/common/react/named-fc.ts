// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactFCWithDisplayName<P = {}> = React.FC<React.PropsWithChildren<P>> & {
    displayName: string;
};

export function NamedFC<P = {}>(
    displayName: string,
    component: React.FC<React.PropsWithChildren<P>>,
): ReactFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactFCWithDisplayName<P>;
}
