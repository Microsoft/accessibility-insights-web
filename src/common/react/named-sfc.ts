// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactSFCWithDisplayName<P = {}> = React.SFC<P> & { displayName: string };

export function NamedSFC<P = {}>(displayName: string, component: React.SFC<P>): ReactSFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactSFCWithDisplayName<P>;
}
