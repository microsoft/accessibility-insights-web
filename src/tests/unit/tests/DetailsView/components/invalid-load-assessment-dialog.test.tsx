// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Dialog, DialogFooter, PrimaryButton } from '@fluentui/react';
import { render } from '@testing-library/react';
import {
    InvalidLoadAssessmentDialog,
    InvalidLoadAssessmentDialogProps,
} from 'DetailsView/components/invalid-load-assessment-dialog';
import * as React from 'react';
import {
    expectMockedComponentPropsToMatchSnapshots,
    mockReactComponents,
} from '../../../mock-helpers/mock-module-helpers';
jest.mock('@fluentui/react');

describe('InvalidLoadAssessmentDialog', () => {
    mockReactComponents([Dialog, DialogFooter, PrimaryButton]);
    let invalidLoadAssessmentDialogProps: InvalidLoadAssessmentDialogProps;

    beforeEach(() => {
        invalidLoadAssessmentDialogProps = {
            isOpen: true,
            onClose: () => {},
        };
    });

    it('should show when isOpen is set to true', () => {
        const renderResult = render(
            <InvalidLoadAssessmentDialog {...invalidLoadAssessmentDialogProps} />,
        );
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Dialog]);
    });

    it('should not show when isOpen is set to false', () => {
        invalidLoadAssessmentDialogProps.isOpen = false;
        const renderResult = render(
            <InvalidLoadAssessmentDialog {...invalidLoadAssessmentDialogProps} />,
        );
        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
