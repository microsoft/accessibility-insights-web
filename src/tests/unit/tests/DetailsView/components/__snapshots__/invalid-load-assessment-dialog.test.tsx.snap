// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidLoadAssessmentDialog should not show when isOpen is set to false 1`] = `null`;

exports[`InvalidLoadAssessmentDialog should show when isOpen is set to true 1`] = `
<Dialog
  dialogContentProps={
    Object {
      "subText": "Please select a valid *.a11ywebassessment file",
      "title": "Oops! Accessibility Insights canâ€™t open this file.",
      "type": 0,
    }
  }
  hidden={false}
  modalProps={
    Object {
      "containerClassName": "invalidLoadAssessmentDialog",
      "isBlocking": false,
      "onDismissed": [Function],
    }
  }
  onDismiss={[Function]}
>
  <StyledDialogFooterBase>
    <CustomizedPrimaryButton
      data-automation-id="invalid-load-assessment-dialog-ok-button"
      onClick={[Function]}
    >
      OK
    </CustomizedPrimaryButton>
  </StyledDialogFooterBase>
</Dialog>
`;
